phc-rails/app/controllers/api/v1/event_registrations_controller.rb

```diff
@@ -2,94 +2,92 @@ class Api::V1::EventRegistrationsController < ApplicationController
   skip_before_action :verify_authenticity_token
   before_action :verify_security_token
 
+  # Creates an event registration and corresponding account
+  # if it does not exist. Any services that were applied for
+  # are set to applied for the given registration
+  #
+  # @param [String] account_sfid Salesforce ID of Account
+  # @param [Integer] Number__c Unique ID for an event registration
+  # @param [Boolean] *service True if given service was applied for
   def create
-    sf_id = params.delete :account_sfid
-    if sf_id.nil? || sf_id.empty? || sf_id == "\"\""
-      # Create account if salesforce id was not passed in
-      # TODO: Right now, the client basically creates accounts through this endpoint
-      # We should seperate it into 2 API calls, the first to api/v1/create (which is unused right now)
-      params[:sf_id] = nil
-      account = Account.spawn(params)
-    else
-      # We retrieve the corresponding account if the salesforce id was passed in
-      account = Account.find_by(sf_id: sf_id)
+    sf_id =  params[:account_sfid]
+    account = Account.find_by(sf_id: sf_id)
+    account_params = params.permit(Account::API_FIELDS)
+    account_params[:updated] = false
 
-      # An account should always be found for a salesforce ID,
-      # but if not create the account anyway and treat it as a new account
-      if account.nil?
-        account = Account.spawn(params)
-      else
-        account_params = params.permit(Account.fields)
-        account.update account_params
-      end
+    # Update or create account with params
+    if sf_id.blank? || sf_id == '""' || account.nil?
+      account = Account.create(account_params)
+    else
+      account.update(account_params)
     end
-    account.update(updated: false)
-    reg = account.event_registrations.create(Number__c: params[:Number__c])
-    Service.services.each do |name|
-      service = reg.services.create(name: name)
-      # TODO: Make sure we are receiving TRUE as a boolean and not as a string
-      # TODO: Figure out when the heck each case happens and how to test each case
-      if params[name] == true || params[name] == "true"
-        service.applied!
-      end
 
-      # TODO: Don't save every service to speed up queries
-      # if params.include? name
-      #   service = reg.services.create(name: name)
-      #   # TODO: Make sure we are receiving TRUE as a boolean and not as a string
-      #   # TODO: Figure out when the heck each case happens and how to test each case
-      #   if params[name] == true || params[name] == "true"
-      #     service.applied!
-      #   end
-      # end
+    event_reg = account.event_registrations.create(Number__c: params[:Number__c])
+    ServiceList.services.each do |name|
+      event_reg.services.create(name: name).applied! if params[name].to_s == "true"
```

Comment:
should be params[:name]?
also correct me if i'm wrong, but i think all params are strings by default.